File: cpp/digit_problem/countandsay.cpp
class Solution {
public:
    string cas(string str)
    {
        string sret = "";
        
        char last = str[0];
        int count = 1;
        
        for(int i = 1;i < str.length();i++)
        {
            if(str[i] == last)
                count += 1;
            else
            {
                sret += ('0'+count);
                sret += last;
                count = 1;
                last = str[i];
            }
        }
        
        sret += ('0'+count);
        sret += last;
        return sret;
    }

    string countAndSay(int n) {
        if(n == 0)
            return "";
        string sret = "1";
        for(int i = 1;i < n;i++)        
            sret = cas(sret);
        
        return sret;
    }   
};

File: py/Count and Say.py
class Solution:
    # @return a string
    def say(self,s):
        sret = ''
        i = 0
        scur = ''
        
        while i < len(s):
            count = 1
            scur = s[i]
            i += 1
            while i < len(s) and s[i] == scur:
                count += 1
                i += 1
            sret += str(count)
            sret += scur
            
        return sret[:]
        
    def countAndSay(self, n):
        if n == 0:
            return ''
        if n == 1:
            return '1'
            
        count = 1
        s = '1'
        
        while count < n:
            s =  self.say(s)
            count += 1
            
        return s[:]